cryostatio/cryostat-agent
# feat(insights): optionally support Red Hat Insights agent

This PR adds support for the https://github.com/RedHatInsights/insights-java-agent to generate a report and send it to Red Hat Insights. This functionality is only enabled when the Cryostat Operator is deployed with the `INSIGHTS_ENABLED` environment variable set to true, which will never be the case for our upstream builds. This is purely for Red Hat product builds.

To test:
1. Deploy Operator with Insights enabled: `make deploy ENABLE_INSIGHTS=true`
2. Build a sample app with this agent PR
3. Set logging level to debug
4. Observe payload being successfully sent to Insights in the logs

Fixes: https://github.com/cryostatio/cryostat/issues/1763
